### DISCUSSION TITLE
Exporting data & Dimension mismatch in AO integrals overlap

### DISCUSSION THREAD

**Discussion by mmutugi** [QuantumPackage/qp2#350] [Q&A] [answered]
Hello @scemama,

I am trying to extract the AO calculation data from `QP2` for use with another code. I have set the **write** flag for both `io_ao_integrals_overlap` and `io_ao_one_e_integrals` and I see that they are being written in a .gz file in the ezfio directory.  I have also generated `FCIDUMP` files using `qp_run fcidump`. I also saved the 2-electron integrals but I am not sure whether you use 'physicist' or 'chemist' notation (need this for unpacking 8-fold symmetry). I know chemist notation is the default for FCIDUMP files but wanted to make sure.

 
My biggest issue is that I am getting the wrong dimensions on some files, particularly the AO overlap files. For a `n_basis` calculation, I get `n_basis+1` overlaps and 1-electron (core) hamiltonians. I am not sure what is happening there. I noticed that this only happens for some (typically larger) calculations. For example, a C calculation in 6-31g basis yields the correct dimensions but the same calculation in cc-pvdz has one extra basis. Furthermore, I noticed that some of my overlap coefficients were greater than one - is this correct? I have attached an example folder where this has happened. [be.tgz](https://github.com/user-attachments/files/17347711/be.tgz)

I also wanted to ensure that all these files were being written in the same (AO) basis - could you please confirm if this is indeed the case?

Respectfully,
Mark 
Munyi

**Comment by scemama** [comment, marked as answer]
Hello,

The formats used to store the integrals are internal to QP, and it is possible that we change them in the future: we don't guarantee that we will maintain backwards-compatibility for that. So for exporting the integrals, we would rather recommend you to use the TREXIO format:
https://doi.org/10.48550/arXiv.2302.14793
https://github.com/trex-coe/trexio
which is the "stable" way to extract data from QP to be used in other codes. The TREXIo library can be used in Fortran, Python, C, Rust, OCaml, etc... and more and more codes are adopting it.

You can use:
```
qp run export_trexio
```
after having set the name of the trexio file with qp_edit in the `trexio` section.

There is also the trexio-tools repository which contains scripts to manipulate TREXIO files:
https://github.com/TREX-CoE/trexio_tools
for example, there is a converter from TREXIO to FCIDUMP.

Concerning the number of AOs which is different from the number of MOs, the reason is that in QP atomic orbitals are always expressed in cartesian coordinates. When the SCF calculation is made in spherical coordinates, we restrict the SCF to the space spanned by the spherical AOs, but we still work with cartesian AOs for simplicity. This explains the different numbers of AOs and MOs that you can observe. In addition, removing linear dependencies in large basis sets can also lead to fewer MOs.

It is possible that the self-overlap is greater than one if the contraction coefficients in the basis are such that the AO is not normalized.

best,
Anthony
